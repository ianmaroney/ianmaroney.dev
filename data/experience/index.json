{
  "history": [{
    "company": "Truth Collective, LLC",
    "role": "Creative Technologist",
    "start": "2016-10-31",
    "current": true,
    "location": "Rochester, NY",
    "content": "<ul><li>Concepting, scoping, and execution of digital experiences.</li><li>Sourcing, hiring, and oversight of development talent.</li><li>Provisioning, management, and cost-analysis of DevOps infrastructure.</li></ul>"
  }, {
    "company": "Rochester Advertising Federation",
    "role": "Web Chair",
    "start": "2016-07-01",
    "end": "2018-08-01",
    "location": "Rochester, NY",
    "content": "<ul><li>Management of organization website</li><li>Creation and sending of email campaigns</li><li>Creation of event microsites and landing pages.</li></ul>"
  }, {
    "company": "St. John Fisher College",
    "role": "Adjunct Professor, Intro to Web Development",
    "start": "2016-01-01",
    "end": "2016-05-31",
    "location": "Rochester, NY"
  }, {
    "company": "Nazareth College",
    "role": "Adjunct Professor, Intro to Web Development",
    "start": "2015-08-01",
    "end": "2015-12-31",
    "location": "Rochester, NY"
  }, {
    "company": "Makeway, LLC",
    "role": "Co-owner, Director of Development",
    "start": "2011-03-01",
    "end": "2016-10-30",
    "location": "Rochester, NY",
    "content": "<ul><li>Concepting, scoping, and execution of digital experiences.</li><li>Creation and management of custom client back-end systems.</li><li>Sourcing, hiring, and oversight of development talent.</li></ul>"
  }, {
    "company": "Fusion Reactions",
    "role": "Web Director",
    "start": "2010-06-01",
    "end": "2011-03-01",
    "location": "Rochester, NY"
  }, {
    "company": "Local Info Company",
    "role": "Web Developer",
    "start": "2009-09-01",
    "end": "2010-06-01",
    "location": "Rochester, NY"
  }, {
    "company": "Em2 Advertising",
    "role": "Web Developer",
    "start": "2008-05-01",
    "end": "2009-09-01",
    "location": "Rochester, NY"
  }],
  "education": [{
    "company": "Canisius College",
    "role": "BA, Digital Media Arts; Fine Arts Minor",
    "start": "2004-08-01",
    "end": "2008-05-01",
    "location": "Buffalo, NY"
  }]
}
