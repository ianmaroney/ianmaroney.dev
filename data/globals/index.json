{
  "titleExtras": [
    "Developer, DevOps Engineer, and Brand Experience Creator"
  ],
  "connections": [
    {
      "title": "Experience",
      "heading": "My experience",
      "content": "<p>Where I’ve worked over the years and some of my favorite tools to work with.</p>"
    },
    {
      "title": "Contact",
      "heading": "How to find me",
      "content": "<p>Looking to chat? Anything more you’d like to know? Drop me a line any time.</p>"
    },
    {
      "title": "Works",
      "heading": "My work",
      "content": "<p>Some work I’m particularly proud of and some of the brands I’ve worked with along the way.</p>"
    }
  ]
}
