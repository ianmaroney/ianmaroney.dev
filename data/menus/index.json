{
  "main": [
    {
      "title": "Home",
      "url": "/"
    },
    {
      "title": "Experience",
      "url": "/experience"
    },
    {
      "title": "Works",
      "url": "/works"
    },
    {
      "title": "Contact",
      "url": "/contact"
    }
  ]
}
