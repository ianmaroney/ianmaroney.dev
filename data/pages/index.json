{
  "path": "/",
  "slug": "home",
  "title": "Ian Maroney",
  "moduleGroups": [
    {
      "title": "Introduction",
      "modules": [
        {
          "type": "content",
          "hero": true,
          "title": "Intro",
          "heading": "Hey, I’m Ian!<small> A Developer, DevOps engineer, and brand experience creator from Rochester, NY.</small>",
          "content": "<p>Deeply curious, mildly obsessive, and openly anxious; I’ve been helping to create code-backed brand experiences on and off the web for years. Check out some of my <a href='/works'>selected works</a>.</p><p>I chase work that presents a problem to be solved or a facet I can make measurably improved with technology. While I have favorite tools in my bag of tricks, I have experience with a wide range of creative technologies. Have a look at <a href='/experience#history'>my working history</a> or <a href='/experience#technologies'>some of the tech</a> I routinely put to work for brands if you're interested in a bit more detail.</p><p>Anything else you would like to know about me? <a href='/contact#form'>Drop me a line</a> and I’ll most likely respond with an email packed with bulleted lists.</p>"
        }
      ]
    }
  ]
}
