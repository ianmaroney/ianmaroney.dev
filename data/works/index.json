[{
  "client": "Fannie Mae",
  "project": "Campaign Website",
  "hidden": true,
  "links": [{
    "title": "Campaign Website",
    "label": "View demo",
    "url": "https://truthcollective.works/homeownership/kyo/"
  }],
  "content": "<p>Fannie Mae approached Truth Collective, looking for a way to educate the public on the homebuying process in a way would allow people to learn at their own pace, on the go, and without the need for any password-protected gatekeeping.</p><p>We took on the task of writing <em>the</em> book on the process, broke it up into bite-sized pieces, and built in a little fun with quizzes, interactive calculators, video content, and downloadable checklists for when prospective homebuyers start the process out in the world; All machine-translated into 109 languages.</p><p>For my role, I chose a headless WordPress back-end for content management and next.js for a nimble front-end build. The output of that build was designed be exported as a single package and integrated into our clients Drupal environment, allowing for future edits by our clients web teams.</p><h3>Noteworthy features</h3><ul><li>Build process outputs a static website; Capable of being deployed anywhere a typical HTML website can be hosted.</li><li>Content is translated at build-time, and only when it has been updated; Saving clients translations costs and visitors wait-times for translations.</li><li>Code-splitting and pre-loading from next.js allows for snappy load times and seamless transitions between pages.</li><li>An automatic glossary term highlighter puts the spotlight on language that may not be familiar to visitors, allowing them to click to expand a pop-over glossary with the definition of the term.</li><li>Module-based content management means that sections can be quickly edited, reordered, added, or removed.</li></ul>"
}, {
  "client": "Truth Collective",
  "project": "Brand Website",
  "links": [{
    "title": "Brand Website",
    "label": "truthcollective.com",
    "url": "https://truthcollective.com/"
  }],
  "content": "<p>When the time came to build Truth Collective a fresh website, we knew we wanted flexibility, speed, and the opportunity to try out some new tech.</p><p>As the first project in our new AWS infrastructure, we chose to build the website using a headless instance of WordPress to manage content with a custom React front-end. WordPress and the React front-end are run within docker containers within a shared EC2 instance, each kept safe from traffic spikes and geographic latency behind distributed CloudFront proxy-layers.</p><p>A plethora of CSS animations, modular page layouts, and a conversational contact form keep the site feeling fresh, fun, and engaging.</p><p>The react front-end caches content from the WordPress REST API to keep load-times fast and resource usage on the server low.</p><h3>Noteworthy features</h3><ul><li>Cached, server-rendered pages result in quick load times and a strong base for search engine optimization.</li><li>Animated page transitions, scroll-activated reveals, and plenty of modular content types keep the site feeling engaging.</li><li>A WordPress back-end fueled by custom post types and custom fields means that content editors don’t need to write code to make additions or updates to the content.</li></ul>"
}, {
  "client": "Willow Domestic Violence Center",
  "project": "Teen Dating Violence campaign landing page and SMS experience",
  "links": [{
    "title": "Campaign microsite",
    "label": "zerodatingviolence.org",
    "url": "https://zerodatingviolence.org/"
  }, {
    "title": "Campaign SMS experience",
    "label": "+1 (585) 294-0020",
    "url": "tel:+15852940020"
  }],
  "content": "<p>Willow Domestic Violence Center came to Truth Collective, looking for help kick-starting a campaign for Teen Dating Violence Awareness Month. We knew that we would have to reach out to their audience where they spend their time; On their phones.</p><p>Social media content drove visitors to a campaign landing page outfitted with glitchy web animations, quick-hit social media style videos, and emoji-based polling to catch quick-hit reactions.</p><p>To help drive home the reality of what was often described as a non-issue by teens, we crafted an SMS-based multi-day experience to simulate a relationship that shifts from happy and healthy to abusive and dangerous. Students would sign up via text message and receive scripted messages with simple multiple-choice responses over the span of several days. By the end of the experience their digital partner had grown controlling, abusive, and threatening; Illustrating how important it can be to watch out for red flags in a partner.</p><h3>Noteworthy features</h3><ul><li>Landing page filled with code-glitched imagery, social media style videos, and emoji reaction polls.</li><li>SMS experience utilized the Twilio API. Pooled phone numbers are scaled up based on user base.</li><li>Back-end utilizes Feathers JS and MongoDB to track progress.</ul>"
}, {
  "client": "OneSight",
  "project": "Brand Website & Fundraising Portal",
  "links": [{
    "title": "Brand Website",
    "label": "onesight.org",
    "url": "https://onesight.org"
  }, {
    "title": "Fundraising Portal",
    "label": "my.onesight.org",
    "url": "https://my.onesight.org/"
  }],
  "content": "<p>Truth Collective’s client OneSight, a global vision nonprofit, was looking for an overhaul of their digital ecosystem. At their request and within the boundaries of their existing infrastructure, we designed and built a new WordPress website; Offering more modular building blocks to shape content. We also improved the connections to their payment processor, Stripe and their CRM platform, Salesforce.</p><p>We worked with a partner to improve their fundraising platform, built with Ruby on Rails; Making design improvements, user experience enhancements, and bolstering of existing functionality.</p>"
}, {
  "client": "Optic Sky Productions",
  "project": "Brand Website",
  "link": "https://opticskypro.com/",
  "links": [{
    "title": "Brand Website",
    "label": "opticskypro.com",
    "url": "https://opticskypro.com/"
  }],
  "content": "<p>Truth Collective’s friends at Optic Sky Productions wanted a new website after we developed their new visual identity. To fit their brand’s (and team’s) personality we opted for a fluid, multilayered, visually driven experience with plenty of touches of code-driven animation.</p><p>To ensure their ongoing ability to add and update content wouldn’t be hindered, we build the website with WordPress. Ensuring no one on their team would have to write code when making changes.</p>"
}, {
  "client": "Rochester Fringe Festival",
  "project": "Brand Website & Festival Infrastructure",
  "links": [{
    "title": "Brand Website",
    "label": "rochesterfringe.com",
    "url": "https://rochesterfringe.com/"
  }, {
    "title": "Festival Infrastructure",
    "label": "backstage.rochesterfringe.com",
    "url": "https://backstage.rochesterfringe.com/"
  }],
  "content": "<p>The Rochester Fringe Festival came to Makeway looking for an easier way to manage the application, booking, scheduling, and ticketing processes for their 10-day festival in downtown Rochester. Their year-one WordPress solution was incredibly tedious, and put the burden of matching performers with venues on their shoulders.</p><p>To make sure that no party (performer or venue) had an unfair advantage in negotiations, we created a platform that kept details transparent and informative throughout the process, allowing the festival organizers to peek in and make use of performance data to create tickets, keep an up-to-date schedule accessible to the public, and handle the throngs of traffic that would hit the site come festival season.</p><p>Built with the PHP framework CodeIgniter, our back-end platform and front-end website were built to handle large volumes of both data and traffic. The platform and website offered sponsorship and advertising opportunities, maps to venues, and a filtering system for festival-goers to find the right show.</p>"
}]
