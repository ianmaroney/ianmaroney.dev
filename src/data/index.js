import fs from 'fs'

/**
 * Get standardized data as static `pageProps` for a slug-referenced page in `/src/pages`.
 * @param {string} pageSlug The slug for the simple page path to get data for.
 * @param {bool} hasExtras Boolean flag to indicate this page has self-titled extras.
 * @return {object}
 */
export async function getData (pageSlug, hasExtras = false) {
  const rawGlobalData = fs.readFileSync('./data/globals/index.json')
  const globals = JSON.parse(rawGlobalData)

  const rawMenusData = fs.readFileSync('./data/menus/index.json')
  const menus = JSON.parse(rawMenusData)

  const rawPageData = fs.readFileSync(`./data/pages${pageSlug ? `/${pageSlug}` : ''}/index.json`)
  const page = JSON.parse(rawPageData)

  if (page && hasExtras) {
    const rawExtraData = fs.readFileSync(`./data/${pageSlug}/index.json`)
    const extras = JSON.parse(rawExtraData)
    page.extras = { [pageSlug]: extras }
  }

  return {
    props: {
      globals,
      page,
      menus
    }
  }
}
