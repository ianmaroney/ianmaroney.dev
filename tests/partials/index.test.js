/* globals describe, expect, it, beforeAll */

import striptags from 'striptags'
import preloadAll from 'jest-next-dynamic'
import { render, screen, fireEvent } from '@testing-library/react'

import HTMLRender from '@/partials/html-render'
import Nav from '@/partials/nav'
import Meta from '@/partials/meta'
import ModuleGroup from '@/partials/module-group'
import ModuleGroups from '@/partials/module-groups'
import Form from '@/partials/form'

import data from '../data'

beforeAll(async () => {
  await preloadAll()
})

describe('Partials HTMLRender', () => {
  it('null HTMLRender', async () => {
    render(<HTMLRender />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('null HTMLRender: tag', async () => {
    render(<HTMLRender tag={data.contentTag} />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('HTMLRender: content', async () => {
    render(<HTMLRender content={data.contentInTag} />)
    expect(screen.getByText(striptags(data.contentInTag))).toBeInTheDocument()
  })

  it('HTMLRender: content + tag', async () => {
    render(<HTMLRender content={data.content} tag={data.contentTag} />)
    expect(screen.getByText(data.content)).toBeInTheDocument()
  })

  it('HTMLRender: content + tag + tagAttr{onClick}', async () => {
    render(<HTMLRender content={data.content} tag={data.contentTag} tagAttr={data.contentTagAttrOnClick} />)
    expect(screen.getByText(data.content)).toBeInTheDocument()
    fireEvent.click(screen.getByText(data.content))
    expect(data.contentTagAttrOnClick.onClick).toHaveBeenCalledTimes(1)
  })

  it('HTMLRender: content + tag + tagAttr{className}', async () => {
    const tagAttr = Object.assign({}, data.contentTagAttrClass)
    const { container } = render(<HTMLRender content={data.content} tag={data.contentTag} tagAttr={tagAttr} />)
    expect(screen.getByText(data.content)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('HTMLRender: content + tag + tagAttr{className, ref}', async () => {
    const tagAttr = Object.assign({}, data.contentTagAttrClassRef)
    const { container } = render(<HTMLRender content={data.content} tag={data.contentTag} tagAttr={tagAttr} />)
    expect(screen.getByText(data.content)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})

describe('Partials Form', () => {
  it('null Form', async () => {
    render(<Form />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('null Form: success', async () => {
    render(<Form success={Object.assign({}, data.formData.success)} />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('null Form: error', async () => {
    render(<Form error={Object.assign({}, data.formData.error)} />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('null Form: success + error', async () => {
    render(<Form success={Object.assign({}, data.formData.success)} error={Object.assign({}, data.formData.error)} />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('Form: fields', async () => {
    const fields = [...data.formData.fields]
    render(<Form fields={fields} />)
    expect(screen.getByLabelText(fields[0].title)).toBeInTheDocument()
    expect(screen.getByLabelText(fields[1].title)).toBeInTheDocument()
    expect(screen.getByLabelText(fields[2].title)).toBeInTheDocument()
  })

  it('Form: fields + success', async () => {
    const fields = [...data.formData.fields]
    render(<Form fields={fields} success={Object.assign({}, data.formData.success)} />)
    expect(screen.getByLabelText(fields[0].title)).toBeInTheDocument()
    expect(screen.getByLabelText(fields[1].title)).toBeInTheDocument()
    expect(screen.getByLabelText(fields[2].title)).toBeInTheDocument()
  })

  it('Form: fields + error', async () => {
    const fields = [...data.formData.fields]
    render(<Form fields={fields} error={Object.assign({}, data.formData.error)} />)
    expect(screen.getByLabelText(fields[0].title)).toBeInTheDocument()
    expect(screen.getByLabelText(fields[1].title)).toBeInTheDocument()
    expect(screen.getByLabelText(fields[2].title)).toBeInTheDocument()
  })

  it('Form: fields success + error', async () => {
    const fields = [...data.formData.fields]
    render(<Form fields={fields} success={Object.assign({}, data.formData.success)} error={Object.assign({}, data.formData.error)} />)
    expect(screen.getByLabelText(fields[0].title)).toBeInTheDocument()
    expect(screen.getByLabelText(fields[1].title)).toBeInTheDocument()
    expect(screen.getByLabelText(fields[2].title)).toBeInTheDocument()
  })
})

describe('Partials Meta', () => {
  it('null Head', async () => {
    render(<Meta />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('null Head: globals', async () => {
    render(<Meta globals={Object.assign({}, data.metaGlobals)} />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  // TODO: Can the <Meta /> component be tested with data? Does its unique intent prevent it from rendering directly into the DOM?
  // it('Head: page', async () => {
  //   render(<Nav page={metaPage} />)
  //   expect(screen.getByText(metaPage.title)).toBeInTheDocument()
  // })
})

describe('Partials ModuleGroup', () => {
  it('null ModuleGroup', async () => {
    render(<ModuleGroup />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('null ModuleGroup: extras', async () => {
    const extras = Object.assign({}, data.moduleGroupExtras)
    render(<ModuleGroup extras={extras} />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('ModuleGroup: moduleGroup', async () => {
    const moduleGroup = Object.assign({}, data.moduleGroupModuleGroup)
    moduleGroup.modules[0].type = 'content'
    render(<ModuleGroup moduleGroup={moduleGroup} />)
    expect(screen.getByText(moduleGroup.modules[0].heading)).toBeInTheDocument()
  })

  it('ModuleGroup: moduleGroup + extras', async () => {
    const moduleGroup = Object.assign({}, data.moduleGroupModuleGroup)
    moduleGroup.modules[0].type = 'experience'
    const extras = Object.assign({}, data.moduleGroupExtras)
    render(<ModuleGroup moduleGroup={moduleGroup} extras={extras} />)
    expect(screen.getByText(moduleGroup.modules[0].heading)).toBeInTheDocument()
    expect(screen.getByText(extras.experience.content[0].company)).toBeInTheDocument()
  })
})

describe('Partials ModuleGroups', () => {
  it('null ModuleGroups', async () => {
    render(<ModuleGroups />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('ModuleGroups: page', async () => {
    const page = Object.assign({}, data.moduleGroupsPage)
    render(<ModuleGroups page={page} />)
    expect(screen.getByText(page.moduleGroups[0].modules[0].heading)).toBeInTheDocument()
  })

  it('ModuleGroups: page w/extras', async () => {
    const page = Object.assign({}, data.moduleGroupsPage)
    page.moduleGroups[0].modules[0].type = 'experience'
    page.moduleGroups[0].modules[0].title = 'Content'
    render(<ModuleGroups page={page} />)
    expect(screen.getByText(page.moduleGroups[0].modules[0].heading)).toBeInTheDocument()
  })
})

describe('Partials Nav', () => {
  it('null Nav', async () => {
    render(<Nav />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('null Nav: setNavigating', async () => {
    render(<Nav setNavigating={data.navHandleClick} />)
    expect(screen.queryByText(/\w/)).toBeNull()
  })

  it('Nav: navItems', async () => {
    const navItems = [...data.navNavItems]
    render(<Nav navItems={navItems} />)
    expect(screen.getByText(navItems[0].title)).toBeInTheDocument()
  })

  it('Nav: navItems + handleClick', async () => {
    const navItems = [...data.navNavItems]
    render(<Nav navItems={navItems} handleClick={data.navHandleClick} />)
    expect(screen.getByText(navItems[0].title)).toBeInTheDocument()

    fireEvent.click(screen.getByText(navItems[0].title))
    expect(data.navHandleClick).toHaveBeenCalledTimes(1)
  })
})
