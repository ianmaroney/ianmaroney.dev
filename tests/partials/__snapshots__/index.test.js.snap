// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partials HTMLRender HTMLRender: content + tag + tagAttr{className, ref} 1`] = `
<div>
  <p
    class="content"
  >
    Content.
  </p>
</div>
`;

exports[`Partials HTMLRender HTMLRender: content + tag + tagAttr{className} 1`] = `
<div>
  <p
    class="content"
  >
    Content.
  </p>
</div>
`;
